### PR TITLE
Add base code for doctor's name implementation

### PR DESCRIPTION
**Pull request by aldentantan**

### CODE REVIEW

File: src/main/java/seedu/address/model/person/DoctorName.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof Name)) {
```
Comment (Johnwz123) [inline src/main/java/seedu/address/model/person/DoctorName.java:54]:
```suggestion
        if (!(other instanceof DoctorName)) {
```

File: src/main/java/seedu/address/model/person/DoctorName.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            return false;
+        }
+
+        Name otherName = (Name) other;
```
Comment (Johnwz123) [inline src/main/java/seedu/address/model/person/DoctorName.java:58]:
```suggestion
        DoctorName otherName = (DoctorName) other;
```

**Comment by aldentantan** [comment]
Closes #49.

**Review by Johnwz123 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**